### PR TITLE
Enable all images on ppc64le and s390x

### LINKED ISSUE #2468: Support for Ansible Operator image on s390x and ppc64le architectures
## Feature Request

**Is your feature request related to a problem? Please describe.**
Today Ansible Operators generated with the operator-sdk can not be run on the s390x and ppc64le architectures. This reduces the portability of operators designed in this way.

I pinpointed the compatibility to three core issues:

1. The ansible scaffold calls out the `x86` package for `inotify` ( there is  a comment about  removing `inotify`, which would clear this  point).
https://github.com/operator-framework/operator-sdk/blob/6b4e1370f18e5d028b283375b740026d2efb0a44/internal/scaffold/ansible/dockerfilehybrid.go#L68

2. `libffi-devel` and  `openssl-devel` are not  installed in the `s390x` and `ppc64le` versions of the docker file build. This  should  go away if `inotify` is removed.
https://github.com/operator-framework/operator-sdk/blob/6b4e1370f18e5d028b283375b740026d2efb0a44/internal/scaffold/ansible/dockerfilehybrid.go#L66

3. The ansible scaffold calls out the `x86` binary for `tini`. This actually might also be a problem, because the scaffold is pointing at the latest release.
https://github.com/operator-framework/operator-sdk/blob/6b4e1370f18e5d028b283375b740026d2efb0a44/internal/scaffold/ansible/dockerfilehybrid.go#L90

Additionally, in the case of `s390x` there's an additional caveat in that the  `operator-sdk` is not currently built for that architecture. In testing I applied the fixes from  PR  #2329 to my environment and  that allowed me to build the `s390x` image.

**Describe the solution you'd like**
To resolve this there are two general changes:
1. Merge the changes outlined by PR #2329 to support the `s390x` build.
2. Add a means to change image targets based on architecture to `internal/scaffold/ansible/dockerfilehybrid.go`

I've applied brute force changes to `dockerfilehybrid.go` in my fork to generate some functional images for `s390x` and  `ppc64le`:

*  [s390x changes](https://github.com/mew2057/operator-sdk/blob/0293feee07d40e653eed7619d0812bf44c6b1a4d/internal/scaffold/ansible/dockerfilehybrid.go#L66)
* [ppc64le changes](https://github.com/mew2057/operator-sdk/blob/490284b6912c3f1d0ae1821aeb559c5c2b8a54b0/internal/scaffold/ansible/dockerfilehybrid.go#L66)
* [quay repo with generated images](https://quay.io/repository/mew2057/ansible-operator?tab=tags)

### PR DESCRIPTION
**Pull request by yselkowitz**
This replaces #2329 and should fix #2468.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -164,6 +168,9 @@ image-push: image-push-ansible image-push-helm image-push-scorecard-proxy ## Pus
 image-push-ansible:
 	./hack/image/push-image-tags.sh $(ANSIBLE_BASE_IMAGE):dev $(ANSIBLE_IMAGE)
```
Comment (joelanford) [inline Makefile]:
```suggestion
	./hack/image/push-image-tags.sh $(ANSIBLE_BASE_IMAGE):dev $(ANSIBLE_IMAGE)-$(shell go env GOARCH)
```

File: Makefile [hunk 2]
```diff
@@ -173,6 +180,9 @@ image-push-helm-multiarch:
 image-push-scorecard-proxy:
 	./hack/image/push-image-tags.sh $(SCORECARD_PROXY_BASE_IMAGE):dev $(SCORECARD_PROXY_IMAGE)
```
Comment (joelanford) [inline Makefile]:
```suggestion
	./hack/image/push-image-tags.sh $(SCORECARD_PROXY_BASE_IMAGE):dev $(SCORECARD_PROXY_IMAGE)-$(shell go env GOARCH)
```

File: internal/scaffold/ansible/dockerfilehybrid.go [hunk 3]
```diff
@@ -64,16 +64,17 @@ ENV OPERATOR=/usr/local/bin/ansible-operator \
 # yum clean all && rm -rf /var/yum/cache/* first
 RUN yum clean all && rm -rf /var/cache/yum/* \
  && yum -y update \
- && yum install -y python36-devel gcc python3-pip python3-setuptools \
+ && yum install -y libffi-devel openssl-devel python36-devel gcc python3-pip python3-setuptools \
```
Comment (joelanford) [inline internal/scaffold/ansible/dockerfilehybrid.go]: Just curious why these are being added?

File: Makefile (hunk 1 above)
Comment (yselkowitz) [reply Makefile]: Thanks, revised accordingly.

File: Makefile (hunk 2 above)
Comment (yselkowitz) [reply Makefile]: Thanks, revised accordingly.

File: internal/scaffold/ansible/dockerfilehybrid.go (hunk 3 above)
Comment (yselkowitz) [reply internal/scaffold/ansible/dockerfilehybrid.go]: Some Python extensions (e.g. cffi) have "manylinux" wheels for x86 but not for other arches, resulting in them needing to be built from source.  These were the additional dependencies I encountered when working on implementing this PR.

**Comment by joelanford** [comment]
@yselkowitz Since `deploy` phases run only on `master` and tag builds, let's keep an eye out to make sure everything passes on this `master` build.

**Comment by yselkowitz** [comment]
@joelanford looks like all three images' master builds from a few minutes ago are three-arch manifest listed, as expected.  Anything else that needs to be checked?

IBM is clamoring for these builds.  If this was indeed successful, would it be a candidate for `/cherry-pick v0.15.x`?  If not, when might we expect v0.16.0?

**Comment by clnperez** [comment]
Thanks again @yselkowitz. 

@joelanford -- any comment on that release date or cherry-pick? ^

**Comment by yselkowitz** [comment]
@clnperez v0.16.0 has been released.

**Pull request merged**